src/Data/DataProviderInterface.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Data;
+
+use Yiisoft\Data\Paginator\PaginatorInterface;
+
+/**
+ * DataProviderInterface is the interface that must be implemented by data provider classes.
+ *
+ * Data providers are components that sort and paginate data, and provide them to widgets such as
+ * {@see Yiisoft\Yii\DataView\GridView}, {@see Yiisoft\Yii\DataView\ListView}.
+ *
+ * For more details and usage information on DataProviderInterface, see the
+ * [guide article on data providers](guide:output-data-providers).
+ */
+interface DataProviderInterface
```

Comment:
How about using interfaces from `yiisoft/data` instead of having a special one?

Reply:
The problem is that for now most of them are incompatible, but I can do it in another PR, it is easier to improve everything, when it is already working.